Products/Zuul/facades/devicefacade.py

```diff
@@ -953,9 +954,12 @@ def getComponentGraphs(self, uid, meta_type, graphId, allOnSame=False):
             return []
 
         if allOnSame:
-            # ZEN-26498 identify by name to match individual charts
+            # ZEN-26498 ZEN-27126 get docker legends to match individual graphs
             for comp in components:
-                comp.id = comp.name()
+                if (hasattr(comp,'docker_host')):
```

Comment:
I suggest looking at the monkeypatch technique used in https://github.com/zenoss/ZenPacks.zenoss.vSphere/blob/develop/ZenPacks/zenoss/vSphere/patches/platform.py and https://github.com/zenoss/ZenPacks.zenoss.Microsoft.Windows/blob/develop/ZenPacks/zenoss/Microsoft/Windows/patches/platform.py as an alternative that doesn't taint DeviceFacade with docker-specific knowledge. This would apply to the Docker ZP.